Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM python:3.13-slim AS builder
+COPY --from=ghcr.io/astral-sh/uv:latest /uv /uvx /bin/
+
+# Change the working directory to the `app` directory
+WORKDIR /app
+
+# Install dependencies
+RUN --mount=type=cache,target=/root/.cache/uv \
+  --mount=type=bind,source=uv.lock,target=uv.lock \
+  --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+  uv sync --frozen --no-install-project --no-editable
+
+# Copy the project into the intermediate image
+ADD . /app
+
+# Sync the project
+RUN --mount=type=cache,target=/root/.cache/uv \
+  uv sync --frozen --no-editable
+
+FROM python:3.13-slim
+
+# Copy the environment, but not the source code
+COPY --from=builder --chown=app:app /app/.venv /app/.venv
```

Comment:
curious why its better to do it this way?

Reply:
I followed (*cough* copied) the example [here](https://docs.astral.sh/uv/guides/integration/docker/#non-editable-installs). Because we've done a non-editable install, we don't need to include the code itself in the final image, we can just copy the venv directly